### ISSUE TITLE
SignalR AspNetCore only working with Kestrel (OutOfProcess) and not IIS (InProcess), getting 404 on negotiate

### ISSUE THREAD

**Issue by BRB2000** [aspnet/AspNetCore#14837]
I have an AspNetCore 3 WebAPI with SignalR setup. I am getting a 404 response on the hubconnection.StartAsync() method of the client. The client has version 3.0 final. Whenever I switch to OutOfProcess hosting using Kestrel and IIS as reverse proxy, everything starts working magically. This is occuring both on two of my developement machines using IIS Express and on my production server farm using IIS 10.0.

The log of the InProcess hosted app shows the 404 response:
`POST /SignalR/negotiate - 443 - 91.6.---.--- Microsoft.AspNetCore.Http.Connections.Client/3.0.0 - 404 0 0 99`

**Comment by anurse** [comment]
We do test with in-proc and aren't aware of any specific issues with it. Can you provide a runnable sample project that reproduces the issue?

The `negotiate` request is a plain HTTP request (there's no WebSockets or other funky stuff at that point), so it's more likely this is a routing problem of some kind. If you can't share a full project, can you share your `Program.cs` and `Startup.cs` files?

**Comment by anurse** [comment]
Also, if you can collect server-side logs (not just the IIS logs, but actual logs from the ASP.NET Core app) that might help. See our [Diagnostics Guide](https://docs.microsoft.com/en-us/aspnet/core/signalr/diagnostics?view=aspnetcore-3.0) for instructions on how to configure the logging.

**Comment by BRB2000** [comment]
Oh my god, I just made a fool of myself :). Turns out, that I made a very silly mistake, which in combination with the different handling in Kestrel with reverse-proxy and IIS AspNetCoreModule led me on the wrong track.
So, the issue was, that my endpoint URL on the client was set to `https://api.example.com//signalR`, instead of `https://api.example.com/signalR`, which apparently gets handled differently in the two hosting scenarios.
Thank you very much for your efforts, I appreciate the support. This is by far the best experience I had with filing an issue on GitHub.
